discovery_child_development/utils/google_utils.py

```diff
@@ -79,3 +84,125 @@ def create_client() -> bigquery.Client:
     # Initialize a client with the provided credentials
     client = bigquery.Client(credentials=credentials)
     return client
+
+
+def write_like_condition(term: Union[str, List[str]], table: str, field: str) -> str:
+    """Create a LIKE condition for a search term or a list of search terms"""
+    if len(term) == 1:
+        return f'{table}.{field} LIKE "%{term[0]}%"'
+    else:
+        # Write an AND condition if more than one search term
+        return "(" + " AND ".join([f'{table}.{field} LIKE "%{t}%"' for t in term]) + ")"
+
+
+def create_patents_query(search_terms: List[str]) -> str:
+    """Create a query to fetch data from BigQuery using search terms:
+    the query checks search terms in the title and abstract
+
+    Args:
+    search_terms (List[str]): list of search terms
+
+    Returns
+    str: query to fetch data from BigQuery
+    """
+
+    # Create a list of 'LIKE' conditions for each search term
+    like_conditions = {}
+    for field in ["title", "abstract"]:
+        like_conditions[field] = [
+            write_like_condition(term, "gpr", field) for term in search_terms
+        ]
+
+    # Join conditions with 'OR'
+    combined_conditions_title = " OR ".join(like_conditions["title"])
+    combined_conditions_abstract = " OR ".join(like_conditions["abstract"])
+
+    q = f"""
+    WITH
+    pubs as (
+        SELECT DISTINCT
+            pub.publication_number
+        FROM `patents-public-data.patents.publications` pub
+            INNER JOIN `patents-public-data.google_patents_research.publications` gpr ON
```

Comment:
Would you be able to document somewhere why it's necessary to inner join these specific tables - why we need only publications that have appeared in both tables?

Is it the case that the `google_patents_research.publications` table has abstracts translated into English, while `patents.publication` has the grant date?

Reply:
Yes that's the reason - I'll add it in the docstring